### PR TITLE
fix(highlights): Ensure that Highlights are refreshed on init.

### LINKED ISSUE #3410: If the Highlights section is re-enabled, no cards are displayed until the page is refreshed
https://bugzilla.mozilla.org/show_bug.cgi?id=1398779

[Affected versions]:
- Firefox 57.0a1 Build ID 20170910220126

[Affected Platforms]:
- All Windows
- All Mac
- All Linux

[Prerequisites]:
- browser.newtabpage.activity-stream.enabled is set to true in about:config on a clean new profile.
- Have enough history so that Top Sites and Highlights are populated.

[Steps to reproduce]:
1. Start the browser with the profile from prerequisites.
2. Click the "Customize your New Tab page" button.
3. Disable the Highlights section.
4. Re-enable the Highlights section.

[Expected results]:
- All previously hidden Highlights are displayed.

[Actual results]:
- Three empty placeholders are displayed.

[Notes]:
- Attached a screen recording of the issue:
highlights not displayed after re-enabling them

### PR DESCRIPTION
**Pull request by AdamHillier**
Closes #3410.

### CODE REVIEW

File: system-addon/lib/HighlightsFeed.jsm [hunk 1]
```diff
@@ -41,13 +41,26 @@ this.HighlightsFeed = class HighlightsFeed {
...
 
   uninit() {
     SectionsManager.disableSection(SECTION_ID);
   }
 
   async fetchHighlights(broadcast = false) {
+    // We need TopSites to have been initialised for deduping
+    if (!this.store.getState().TopSites.initialized) {
+      await new Promise(resolve => {
+        const unsubscribe = this.store.subscribe(async () => {
```
Comment (rlr) [inline system-addon/lib/HighlightsFeed.jsm]: is this `async` required since there are no `await`s inside? I am still an async/await newb so I'm not sure.

Comment (AdamHillier) [reply system-addon/lib/HighlightsFeed.jsm]: Ah no it's not that's a good point, cheers

**Comment by as-pine-proxy** [comment]
[Treeherder: ae8d646ee0ce417e03bbf97755556b7c4a704ced](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=ae8d646ee0ce417e03bbf97755556b7c4a704ced)

**Comment by Mardak** [comment]
Did you end up pushing this to try to check talos?

**Pull request merged**